pulp_container/app/cache.py

```diff
@@ -0,0 +1,132 @@
+from aiohttp.web_exceptions import HTTPForbidden
+
+from django.core.exceptions import ObjectDoesNotExist
+
+from pulpcore.plugin.cache import CacheKeys, AsyncContentCache, SyncContentCache
+
+from pulp_container.app.models import ContainerDistribution
+from pulp_container.app.exceptions import RepositoryNotFound
+
+ACCEPT_HEADER_KEY = "accept_header"
+
+
+class RegistryCache:
+    """A class that overrides the default key specs."""
+
+    def __init__(self, base_key=None, expires_ttl=None, auth=None):
+        """Initialize the parent class with the plugin's specific keys."""
+        updated_keys = (CacheKeys.path, CacheKeys.method, ACCEPT_HEADER_KEY)
+        super().__init__(base_key=base_key, expires_ttl=expires_ttl, keys=updated_keys, auth=auth)
+
+
+class RegistryContentCache(RegistryCache, AsyncContentCache):
+    """A wrapper around the Redis content cache handler tailored for the content application."""
+
+    def make_key(self, request):
+        """Make a key composed of the request's path, method, host, and accept header."""
+        accept_header = ",".join(sorted(request.headers.getall("accept", [])))
```

Comment:
I understand that `getall` will get all of the occurrences of `Accept` header, which can be multiple,however, every Accept header can contain a comma separated list of headers.
see more for the reference https://github.com/pulp/pulp_container/blob/main/pulp_container/app/utils.py#L16

Reply:
As far as I know, this had to be made because of how `MultiDict` stores its keys. I focused only on Accept headers that looked like this:
```
Accept: application/vnd.docker.distribution.manifest.list.v2+json,application/vnd.docker.distribution.manifest.v2+json
```

The call converts the header back to its original value from:
```
<CIMultiDictProxy( 'Accept': 'application/vnd.docker.distribution.manifest.list.v2+json',  'Accept': 'application/vnd.docker.distribution.manifest.v2+json')>
```

Are you sure a user can pass multiple Accept headers to a request?

Reply:
based on the http rfc i read time ago - yes, one can pass multiple headers in a request